src/utils/setCommentForPath.ts

```diff
@@ -0,0 +1,20 @@
+import { TreeCore } from "../tree/types";
+/**  Replaces the existing comment by a given string
+ * @param {TreeCore} treeCore the whole tree
+ * @param {string} path The path of a specific file
+ * @param {string} comment any string
+ * @returns {void}
+ */
+
+export const setCommentForPath = (treeCore:TreeCore, path: string, comment: string): void => {
+    treeCore.forEach((core) => {
+        if (path=core.text) {
+            core.comment=comment;
+        }
+    }
```

Comment:
If the core.path === path passed in, then you know which path to update their comment for. The way you are doing it looks good!

Your test function will have a TreeCore Before and a TreeCore After to make sure the correct Core was updated wit hthe Correct Comment.

Reply:
@ralph-dev 
The problem is I think this will only change the comment locally...
I have updated the codes and in my test I see the correct result with console.log but outside of it, nothing changes

Reply:
Yes but that is the point because someone using this would want to change only his comment for a specific instance.

Reply:
@ralph-dev but how should I write test for it? Since outside of the function the value doesn't change?

Reply:
The value changes in the TreeObject you may need a For Of loop instead of For Each for this btw. You can choose to also instead change the function signature to return a new TreeCore and display that in the UI instead :)

Reply:
@ralph-dev actually this is old... I didn't use foreach or for loop... I'm using findIndex in the new code

Reply:
Ok that's fine then make sure this returns a new TreeCore object instead then it will work / be testable